doc/feature/compartmentReport.md

```diff
@@ -0,0 +1,422 @@
+ZeroEQ restrictions at the same time.
+
+This specification proposes a new interface for the CompartmentReport with
+these goals in mind.
+
+## API
+
+    /** Read & write access to a CompartmentReport.
+     *
+     * The supported types are binary (r), HDF5 (rw) and stream (rw) reports.
+     *
+     * A compartment report is a discrete sampling in time and space of a
+     * continuous scalar variable over a cell population. Time is discretized
+     * in frames and space is discretized in compartments. Compartments are
+     * subdivisions of cells' sections, each section is divided up into n
+     * equally sized compartments longitudinally. A frame is the sampling of
+     * the values that the report variable takes at each compartment at a
+     * precise point in time. Frames are evenly spaced by the report timestep
+     * (ts). In the simplest interpolation scheme, let f(t) = x be the value
+     * of the report variable at a compartment at frame n (t = n * ts), f(t +
+     * e) = x for every e in [0, timestep).
+     *
+     * Following RAII, this class is ready to use after the creation and will
+     * ensure release of resources upon destruction.
+     *
+     * Thread-safey is only guaranteed for const access. That also means that
+     * accessing the const& types returned by some functions is not thread-safe
+     * while a non const operation is called or any returned future is being
+     * waited.
+     */
+    CompartmentReport : public boost::noncopyable
+    {
+    public:
+
+        ~CompartmentReport();
+
+        /** Open a compartment report
+         *
+         * In read mode, this function may block if needed until all the
+         * information has been collected (e.g. the report metadata).
+         */
+        CompartmentReport( const URI& uri, const int mode )
+
+        /*
+         * Open a report in read mode with a subset selection.
+         *
+         * This function may block if needed until all the information has
+         * been collected (e.g. the report metadata).
+         *
+         * @param uri
+         * @param subset The set of gids to be reported. This set should be
+         *        understood as a filter.
+         */
+        CompartmentReport( const URI& uri, const GIDSet& subset );
+
+        /** Update compartment mapping wrt the given GIDs.
+         *
+         * Change the cell set of interest in an report open in read mode. The
+         * GIDSet will become effective once the next read operation is
+         * completed.
+         *
+         * This operation may throw away cached data and make getCurrentTime()
+         * jump a step bigger than the timestep for the next read. It is
+         * implementation defined when this may happen.
+         *
+         * @throw std::runtime_error is the report was open in write mode or
+         *        there's an on-going read operation.
+         */
+        void updateMapping( const GIDSet& gids );
```

Comment:
Maybe we should find a different name which does not include mapping. This is confusing with the writeMapping() function below where mapping has a different semantics.